pilz_teleoperation/test/integrationtests/tst_key_teleop.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python
+# Copyright (c) 2019 Pilz GmbH & Co. KG
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import rospy
+import unittest
+
+from geometry_msgs.msg import TwistStamped
+from control_msgs.msg import JointJog
+
+
+class TestKeyTeleoperation(unittest.TestCase):
+    """
+    Test that the key_teleoperation file brings up the teleoperation driver and calls its update method
+    """
+    def __init__(self, *args, **kwargs):
+        super(TestKeyTeleoperation, self).__init__(*args, **kwargs)
+        self.twist_published = False
+
+    def setUp(self):
+        rospy.loginfo("SetUp called...")
+
+    def tearDown(self):
+        rospy.loginfo("TearDown called...")
+
+    def callback(self, msg):
+        self.twist_published = True
+
+    def test_startup(self):
+        """ Test if driver gets started """
+        subscriber_twist = rospy.Subscriber('/jog_server/delta_jog_cmds', TwistStamped, self.callback)
+        subscriber_joint = rospy.Subscriber('/jog_server/joint_delta_jog_cmds', JointJog, self.callback)
```

Comment:
This simply tests publication, right? I'm not sure if that is worth a self-written test file; could be accomplished using [hztest](http://wiki.ros.org/rostest/Nodes) as well... But probably this is just a starting point and will add more checks in the future??

Reply:
yes, it was just a simple way to test the test structure and do anything useful in the meantime. This should be improved in the future. I would however suggest we don't do it in this pull request, to keep the merge small and have the test structure in the main branch already